java-client/src/main/java/com/wavefront/integrations/WavefrontDirectSender.java

```diff
@@ -111,26 +215,29 @@ static String pointToString(String name, double value, @Nullable Long timestamp,
 
   @Override
   protected void internalFlush() throws IOException {
-
     if (!isConnected()) {
         return;
     }
+    internalFlush(pointBuffer, PUSH_FORMAT_GRAPHITE_V2);
+    internalFlush(histogramBuffer, PUSH_FORMAT_HISTOGRAM);
+  }
 
-    List<String> points = getPointsBatch();
+  private void internalFlush(LinkedBlockingQueue<String> buffer, String format) throws IOException {
+    List<String> points = getBatch(buffer);
```

Comment:
We currently only flush up to BATCH_SIZE (10k) points per flush. I am wondering whether we should change getBatch to return a `List<List<String>>`. Basically split into batches and flush the entire buffer per flush. Something to run by @panghy .